### PR TITLE
Add trust_remote_code to audio classification CI tests and README.

### PR DESCRIPTION
**Pull request by emascarenhas**
Add trust_remote_code to README and CI tests.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- x Did you make sure to update the documentation with your changes?
- [x ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -584,6 +584,7 @@ def _create_command_line(
                 "--use_habana",
                 "--throughput_warmup_steps 3",
                 "--save_strategy no",
+                "--trust_remote_code True",
```
Comment (regisss) [inline tests/test_examples.py:587]: We shouldn't enable this by default for all tests but rather add it for the tests that need it.
You can add custom arguments for each test within the `extra_arguments` field of the JSON files (e.g. https://github.com/huggingface/optimum-habana/blob/22c069ee7410c0f75822dc690bec6b4d2f016cbc/tests/baselines/wav2vec2_base.json#L14).

Comment (emascarenhas) [reply tests/test_examples.py]: @regisss , I limited it to ast_finetuned_speech_commands_v2.json

**Comment by imangohari1** [comment]
@emascarenhas 
I was looking at this PR. 
- There README file for ac needs another `--trust_remote` in `inference` section. 
- for ac baseline, the test is failing on my end due to perf assertation. 
```diff
diff --git a/examples/audio-classification/README.md b/examples/audio-classification/README.md
index 17935cf1..197bf9ff 100644
--- a/examples/audio-classification/README.md
+++ b/examples/audio-classification/README.md
@@ -174,7 +174,8 @@ python run_audio_classification.py \
     --use_lazy_mode \
     --use_hpu_graphs_for_inference \
     --gaudi_config_name Habana/wav2vec2 \
-    --bf16
+    --bf16 \
+    --trust_remote_code True
 ```

```diff
diff --git a/tests/baselines/ast_finetuned_speech_commands_v2.json b/tests/baselines/ast_finetuned_speech_commands_v2.json
index f518c929..c5c5cb3d 100644
--- a/tests/baselines/ast_finetuned_speech_commands_v2.json
+++ b/tests/baselines/ast_finetuned_speech_commands_v2.json
@@ -8,8 +8,8 @@
                     "learning_rate": 5e-4,
                     "train_batch_size": 32,
                     "eval_accuracy": 0.1871,
-                    "train_runtime": 139.9477,
-                    "train_samples_per_second": 1955.74,
+                    "train_runtime": 141.4154,
+                    "train_samples_per_second": 1751.404,
                     "eval_samples_per_second": 2301.088,
                     "extra_arguments": [
                         "--audio_column_name audio",
```
we could adjust these a bit for this to actually pass. I can't find a RD backed config that matches the our settings for #epocs, bs sizes, etc for this case. The public data are taken from RD backend too, so shouldn't be too bad to adjust here. 
https://developer.habana.ai/get-started/habana-models-performance/

should we make these changes here now, or wait for later? WDYT @regisss ?

With above changes, I get the test to pass 
```bash
GAUDI2_CI=1 RUN_SLOW=true python -m pytest  tests/test_examples.py -s -v -k test_run_audio_classification_ast-finetuned-speech-commands-v2_multi_card
.
.


***** train metrics *****
  epoch                       =         10.0
  max_memory_allocated (GB)   =         3.85
  memory_allocated (GB)       =         3.52
  total_flos                  = 1741424192GF
  total_memory_available (GB) =        94.62
  train_loss                  =        1.021
  train_runtime               =   0:02:21.92
  train_samples_per_second    =     1745.801
  train_steps_per_second      =        6.844
[INFO|trainer.py:1812] 2024-08-16 17:12:58,741 >> Using HPU graphs for inference.
[INFO|trainer.py:1830] 2024-08-16 17:12:58,741 >>
***** Running Evaluation *****
[INFO|trainer.py:1832] 2024-08-16 17:12:58,741 >>   Num examples = 5888
[INFO|trainer.py:1835] 2024-08-16 17:12:58,741 >>   Batch size = 64
100%|██████████| 12/12 [00:02<00:00,  5.89it/s]
***** eval metrics *****
  epoch                       =       10.0
  eval_accuracy               =     0.1921
  eval_loss                   =     4.8415
  eval_runtime                = 0:00:04.57
  eval_samples_per_second     =   2535.925
  eval_steps_per_second       =      5.244
  max_memory_allocated (GB)   =       3.85
  memory_allocated (GB)       =       3.73
  total_memory_available (GB) =      94.62
PASSED

============================================================================================ 1 passed, 55 deselected in 210.54s (0:03:30) ===============

**Comment by emascarenhas** [comment]
> @emascarenhas I was looking at this PR.
> 
> * There README file for ac needs another `--trust_remote` in `inference` section.
> * for ac baseline, the test is failing on my end due to perf assertation.

@imangohari1 ,
Yes lets take this change.
...
> we could adjust these a bit for this to actually pass. I can't find a RD backed config that matches the our settings for #epocs, bs sizes, etc for this case. The public data are taken from RD backend too, so shouldn't be too bad to adjust here. https://developer.habana.ai/get-started/habana-models-performance/
> 

This change isn't required, as I could get a result that would pass with the current setting. See results below.

> should we make these changes here now, or wait for later? WDYT @regisss ?


My results were:
***** train metrics *****
  epoch                       =         10.0
  max_memory_allocated (GB)   =         6.61
  memory_allocated (GB)       =         3.52
  total_flos                  = 1741424192GF
  total_memory_available (GB) =        94.62
  train_loss                  =        1.021
  train_runtime               =   0:02:16.22
  train_samples_per_second    =      1963.21
  train_steps_per_second      =        7.696
[INFO|trainer.py:1812] 2024-08-15 05:10:08,396 >> Using HPU graphs for inference.
[INFO|trainer.py:1830] 2024-08-15 05:10:08,396 >>
***** Running Evaluation *****
[INFO|trainer.py:1832] 2024-08-15 05:10:08,396 >>   Num examples = 5888
[INFO|trainer.py:1835] 2024-08-15 05:10:08,396 >>   Batch size = 64
100%|██████████| 12/12 [00:01<00:00,  7.21it/s]
***** eval metrics *****
  epoch                       =       10.0
  eval_accuracy               =     0.1921
  eval_loss                   =     4.8415
  eval_runtime                = 0:00:04.28
  eval_samples_per_second     =   3084.876
  eval_steps_per_second       =       6.38
  max_memory_allocated (GB)   =       6.61
  memory_allocated (GB)       =       3.73
  total_memory_available (GB) =      94.62

The README change is required, so we should take that change. The other adjustment isn't required I think as our CI systems can get the test to pass without that change.

**Comment by imangohari1** [comment]
> > @emascarenhas I was looking at this PR.
> > 
> > * There README file for ac needs another `--trust_remote` in `inference` section.
> > * for ac baseline, the test is failing on my end due to perf assertation.
> 
> @imangohari1 , Yes lets take this change. ...
> 
> > we could adjust these a bit for this to actually pass. I can't find a RD backed config that matches the our settings for #epocs, bs sizes, etc for this case. The public data are taken from RD backend too, so shouldn't be too bad to adjust here. https://developer.habana.ai/get-started/habana-models-performance/
> 
> This change isn't required, as I could get a result that would pass with the current setting. See results below.
> 
> > should we make these changes here now, or wait for later? WDYT @regisss ?
> 
> My results were: ***** train metrics ***** epoch = 10.0 max_memory_allocated (GB) = 6.61 memory_allocated (GB) = 3.52 total_flos = 1741424192GF total_memory_available (GB) = 94.62 train_loss = 1.021 train_runtime = 0:02:16.22 train_samples_per_second = 1963.21 train_steps_per_second = 7.696 [INFO|trainer.py:1812] 2024-08-15 05:10:08,396 >> Using HPU graphs for inference. [INFO|trainer.py:1830] 2024-08-15 05:10:08,396 >> ***** Running Evaluation ***** [INFO|trainer.py:1832] 2024-08-15 05:10:08,396 >> Num examples = 5888 [INFO|trainer.py:1835] 2024-08-15 05:10:08,396 >> Batch size = 64 100%|██████████| 12/12 [00:01<00:00, 7.21it/s] ***** eval metrics ***** epoch = 10.0 eval_accuracy = 0.1921 eval_loss = 4.8415 eval_runtime = 0:00:04.28 eval_samples_per_second = 3084.876 eval_steps_per_second = 6.38 max_memory_allocated (GB) = 6.61 memory_allocated (GB) = 3.73 total_memory_available (GB) = 94.62
> 
> The README change is required, so we should take that change. The other adjustment isn't required I think as our CI systems can get the test to pass without that change.

Done: https://github.com/huggingface/optimum-habana/pull/1268

**Pull request merged**